SnowWhiteEPFL/edweiss
# ✨ feat: Code Cleanup and Addition of Notifications Page

### This Pull Request introduces the following changes:

- Code Cleanup: Resolve some duplicate code parts - Refactoring of existing components to improve readability, consistency, and maintainability.
- Addition of a Notifications Page: A new page is introduced to centralize notifications, allowing users to view and manage them easily and effectively.

### Context

This PR is part of the ongoing improvements to the application. The goals are to:

- Ensure cleaner, more structured code for future development.
- Enhance user experience by adding a centralized notification feature.

### Changes Made

- Refactoring of Existing Components: Removed redundancies, improved variable names, and restructured files.
- Creation of the Notifications Page:
     - User-friendly interface for displaying and managing notifications.
     - Added styling to ensure a consistent user experience across the application.
     - Possibility to swipe from left to right which allows to mark as read/unread a notification
     - Possibility to swipe from right to left which allows to delete a notification 
     - Firebase function to create and delete a notification 

### Screenshots

Please find below the screenshots showcasing the interface of the new notifications page and the changes made to the code.

#### Light Screen

<img width="443" alt="Screenshot 2024-11-05 at 19 06 23" src="https://github.com/user-attachments/assets/fb0743e2-f942-4edc-9edf-4a1da79971fa">

<img width="445" alt="Screenshot 2024-11-05 at 19 07 41" src="https://github.com/user-attachments/assets/60ab0440-32a2-49d2-b8fe-21161833bf55">

<img width="439" alt="Screenshot 2024-11-05 at 19 08 22" src="https://github.com/user-attachments/assets/1a841eed-9f61-4930-8f1c-40bddaeca1bc">


#### Dark Screen

<img width="440" alt="Screenshot 2024-11-05 at 19 10 28" src="https://github.com/user-attachments/assets/be4e5f61-c37b-4c37-be70-7698d08f8867">


<img width="444" alt="Screenshot 2024-11-05 at 19 09 57" src="https://github.com/user-attachments/assets/b18c4db3-bf5d-4bb5-8242-d5eaa171306b">


### Tests

**Please note that this PR does not provide tests yet.**

### Testing Instructions

1. Run the application locally.
2. Navigate to the new notifications page to verify its correct display.
3. Test the basic interaction on the notifications page (e.g., displaying and updating notifications).

### Remaining Tasks

- Integrate unit and functional tests for the new notifications page.
- Complete code validation once tests are added to ensure optimal functionality of the feature.

### Improvement Tasks

- Possibility to sort the notifications
- Redirect the user to the appropriate screen according to the notification 
- Improve the colour coherency (according to light/dark screen)
- Make the "Read" and "Unread" text field swipe button of the same size

### Additional Notes

- Recall: This PR does not yet include tests. Tests will be added in a future PR.
- Please pay special attention to the refactored code in existing components to identify potential areas for further improvement.
